File: part-02/src/main/java/eu/luminis/workshop/smallsteps/logic/appservice/LegoStockRepository.java
package eu.luminis.workshop.smallsteps.logic.appservice;

import eu.luminis.workshop.smallsteps.logic.domainmodel.valueobjects.LegoStoreId;
import eu.luminis.workshop.smallsteps.logic.domainservice.state.StockState;

public interface LegoStockRepository {
    StockState find(LegoStoreId legoStoreId);

    void save(LegoStoreId legoStoreId, StockState newState);
}

File: part-02/src/main/kotlin/eu/luminis/workshop/smallsteps/logic/appService/LegoStockRepository.kt
package eu.luminis.workshop.smallsteps.logic.appService

import eu.luminis.workshop.smallsteps.logic.domainModel.valueObjects.LegoStoreId
import eu.luminis.workshop.smallsteps.logic.domainService.state.StockState

interface LegoStockRepository {
    fun find(store: LegoStoreId): StockState
    fun save(store: LegoStoreId, newState: StockState)
}